python/tests/mode_coeffs.py

```diff
@@ -36,27 +34,45 @@ def run_mode_coeffs(self, mode_num, kpoint_func):
 
         # mode frequency
         fcen = 0.20  # > 0.5/sqrt(11) to have at least 2 modes
+        df   = 0.5*fcen
 
-        sources = [mp.EigenModeSource(src=mp.GaussianSource(fcen, fwidth=0.5*fcen),
-                                      eig_band=mode_num,
-                                      size=mp.Vector3(0,sy-2*dpml,0),
-                                      center=mp.Vector3(-0.5*sx+dpml,0,0),
-                                      eig_match_freq=True,
-                                      eig_resolution=32) ]
+        source=mp.EigenModeSource(src=mp.GaussianSource(fcen, fwidth=df),
+                                  eig_band=mode_num,
+                                  size=mp.Vector3(0,sy-2*dpml,0),
+                                  center=mp.Vector3(-0.5*sx+dpml,0,0),
+                                  eig_match_freq=True,
+                                  eig_resolution=2*resolution)
 
         sim = mp.Simulation(resolution=resolution,
                             cell_size=cell_size,
                             boundary_layers=boundary_layers,
                             geometry=geometry,
-                            sources=sources,
+                            sources=[source],
                             symmetries=[mp.Mirror(mp.Y, phase=1 if mode_num % 2 == 1 else -1)])
 
         xm = 0.5*sx - dpml  # x-coordinate of monitor
-        mflux = sim.add_mode_monitor(fcen, 0, 1, mp.ModeRegion(center=mp.Vector3(xm,0), size=mp.Vector3(0,sy-2*dpml)))
-        mode_flux = sim.add_flux(fcen, 0, 1, mp.FluxRegion(center=mp.Vector3(xm,0), size=mp.Vector3(0,sy-2*dpml)))
+        mflux = sim.add_mode_monitor(fcen, df, nf, mp.ModeRegion(center=mp.Vector3(xm,0), size=mp.Vector3(0,sy-2*dpml)))
+        mode_flux = sim.add_flux(fcen, df, nf, mp.FluxRegion(center=mp.Vector3(xm,0), size=mp.Vector3(0,sy-2*dpml)))
 
         # sim.run(until_after_sources=mp.stop_when_fields_decayed(50, mp.Ez, mp.Vector3(-0.5*sx+dpml,0), 1e-10))
-        sim.run(until_after_sources=100)
+        sim.run(until_after_sources=200)
+
+        ##################################################
+        # If the number of analysis frequencies is >1, we
+        # are testing the unit-power normalization
+        # of the eigenmode source: we observe the total
+        # power flux through the mode_flux monitor (which
+        # equals the total power emitted by the source as
+        # there is no scattering in this ideal waveguide)
+        # and check that it agrees with the prediction
+        # of the eig_power() class method in EigenmodeSource.
+        ##################################################
+        if nf>1:
+            power_observed=mp.get_fluxes(mode_flux)
+            freqs=mp.get_flux_freqs(mode_flux)
+            power_expected=[source.eig_power(f) for f in freqs]
+            import ipdb; ipdb.set_trace()
```

Comment:
Travis is failing because of this line which should be removed (since it is presumably from debugging?).

Reply:
My bad.